public/app/features/plugins/utils.ts

```diff
@@ -68,3 +79,25 @@ export function buildPluginSectionNav(location: HistoryLocation, pluginNav: NavM
 
   return { main: section, node: activePage ?? section };
 }
+
+// TODO make work for sub pages
+export function getPluginSection(location: HistoryLocation, navIndex: NavIndex, pluginId: string): NavModelItem {
+  // First check if this page exist in navIndex using path, some plugin pages are not under their own section
+  const byPath = navIndex[`standalone-plugin-page-${location.pathname}`];
+  if (byPath) {
+    const parent = byPath.parentItem!;
+    // in case the standalone page is in nested section
+    return parent.parentItem ?? parent;
+  }
+
+  const navTreeNodeForPlugin = navIndex[`plugin-page-${pluginId}`];
+  if (!navTreeNodeForPlugin) {
+    throw new Error('Plugin not found in navigation tree');
+  }
+
+  if (!navTreeNodeForPlugin.parentItem) {
+    throw new Error('Could not find plugin section');
+  }
+
+  return navTreeNodeForPlugin.parentItem;
+}
```

Comment:
Would it make sense to add a few tests for these scenarios (even if only by testing it through `buildPluginSectionNav()`)?
- plugin page is not under its own section (standalone)
- throws an error if the section is not found
- throws an error if the plugin itself is not found in the nav-tree

Reply:
@leventebalogh ah, I just pushed that :+1: